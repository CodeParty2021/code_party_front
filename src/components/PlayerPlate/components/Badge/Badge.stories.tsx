import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Badge } from "./Badge";

export default {
  title: "components/PlayerPlate/Badge",
  component: Badge,
} as ComponentMeta<typeof Badge>;

const Template: ComponentStory<typeof Badge> = (args) => <Badge {...args} />;

export const Turquoise = Template.bind({});
Turquoise.args = {
  badgeName: "宇宙の探究者",
  color: "turquoise",
};

export const Leaf = Template.bind({});
Leaf.args = {
  badgeName: "宇宙の探究者",
  color: "leaf",
};

export const Orange = Template.bind({});
Orange.args = {
  badgeName: "宇宙の探究者",
  color: "orange",
};

export const Magenta = Template.bind({});
Magenta.args = {
  badgeName: "宇宙の探究者",
  color: "magenta",
};
