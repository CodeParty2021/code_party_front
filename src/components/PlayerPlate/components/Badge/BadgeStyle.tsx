import styled, { css } from "styled-components";
import { FONT_WEIGHT } from "styles/constants/constants";
import { PlayerPlateColors } from "components/PlayerPlate/colors";
import { FlexGap } from "styles/FlexGap/FlexGap";

export type BadgeStyleProps = {};

const defaultStyle = css`
  display: flex;
  flex-direction: row;
  align-items: center;
  ${FlexGap({ gap: "16px", direction: "row" })}

  .badge_name {
    transform: matrix(1, 0, -0.08, 1, 0, 0);
    color: ${PlayerPlateColors.badge};
    font-weight: ${FONT_WEIGHT.BOLD};
    font-size: 14px;
    line-height: 20px;
    text-align: center;
  }
`;

export const BadgeStyle = styled.div<BadgeStyleProps>`
  ${defaultStyle}
`;

BadgeStyle.defaultProps = {};
