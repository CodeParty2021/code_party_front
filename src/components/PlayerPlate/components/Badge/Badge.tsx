import { Star } from "components/Star/Star";
import React from "react";
import { BadgeStyle, BadgeStyleProps } from "./BadgeStyle";

type Props = BadgeStyleProps & {
  badgeName?: string;
  color?: "turquoise" | "leaf" | "orange" | "magenta";
};

export const Badge: React.FC<Props> = ({ badgeName, color, ...styleProps }) => {
  return (
    <BadgeStyle {...styleProps}>
      <Star color={color} width="19px" height="19px" />
      <span className="badge_name">{badgeName || "称号なし"}</span>
      <Star color={color} width="19px" height="19px" />
    </BadgeStyle>
  );
};
