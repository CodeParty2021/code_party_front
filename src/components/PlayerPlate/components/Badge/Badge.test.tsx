import React from "react";
import { Badge } from "./Badge";
import "jest-styled-components";
import { render } from "enzyme";

describe("<Badge />", () => {
  it("snapshot test", () => {
    const wrapper = render(<Badge />);

    expect(wrapper).toMatchSnapshot();
  });

  it("render test turquoise", () => {
    expect(
      render(<Badge badgeName="宇宙の探究者" color="turquoise" />)
    ).toBeTruthy();
  });

  it("render test leaf", () => {
    expect(
      render(<Badge badgeName="宇宙の探究者" color="leaf" />)
    ).toBeTruthy();
  });

  it("render test orange", () => {
    expect(
      render(<Badge badgeName="宇宙の探究者" color="orange" />)
    ).toBeTruthy();
  });

  it("render test magenta", () => {
    expect(
      render(<Badge badgeName="宇宙の探究者" color="magenta" />)
    ).toBeTruthy();
  });
});
