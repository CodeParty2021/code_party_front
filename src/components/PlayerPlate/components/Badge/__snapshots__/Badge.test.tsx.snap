// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> snapshot test 1`] = `
.c1 {
  width: 19px;
  height: 19px;
  fill: #F8919F;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 16px;
}

.c0 .badge_name {
  -webkit-transform: matrix(1,0,-0.08,1,0,0);
  -ms-transform: matrix(1,0,-0.08,1,0,0);
  transform: matrix(1,0,-0.08,1,0,0);
  color: #4A5373;
  font-weight: 700;
  font-size: 14px;
  line-height: 20px;
  text-align: center;
}

@supports not (gap:16px) {
  .c0 > *:not(:first-child) {
    margin-left: 16px;
  }
}

<div
  class="c0"
>
  <svg
    class="c1"
    color="pink"
    height="19px"
    width="19px"
    xlink="http://www.w3.org/1999/xlink"
    xmlns="http://www.w3.org/2000/svg"
  />
  <span
    class="badge_name"
  >
    称号なし
  </span>
  <svg
    class="c1"
    color="pink"
    height="19px"
    width="19px"
    xlink="http://www.w3.org/1999/xlink"
    xmlns="http://www.w3.org/2000/svg"
  />
</div>
`;
